### PR TITLE
Fix bug: task was ending when signal was received not after was evaluated

### PR DESCRIPTION
**Pull request by BlueTM**
When was receiving a signal "SIGTERM/SIGHUP" the process immediately exist instead of waiting until gearman current task end.
This fix make it to work as was intended: on signal put a flag on true, before get next task if flag is true stop and exit.

### CODE REVIEW

File: Service/GearmanExecute.php [hunk 1]
```diff
@@ -90,22 +90,12 @@ public function __construct(GearmanCacheWrapper $gearmanCacheWrapper, array $def
...
-        }
     }
 
     /**
-     * Toggles that work should be stopped, we only subscribe to SIGTERM and SIGHUP
-     * @param int $signo Signal number
+     * Toggles that work should be stopped, we only subscribe to SIGTERM,SIGINT and SIGHUP
      */
-    public function handleSystemSignal($signo)
+    public function handleSystemSignal(int $signo): void
```
Comment (draage2572) [inline Service/GearmanExecute.php:98]: no need for signo, it's not used anywhere and the method is not declared int the abstract

File: Command/GearmanJobExecuteCommand.php [hunk 2]
```diff
@@ -79,6 +79,16 @@ protected function configure()
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
+        if (extension_loaded('pcntl') && function_exists('pcntl_async_signals')) {
```
Comment (draage2572) [inline Command/GearmanJobExecuteCommand.php:82]: extension loaded should be enough, no need to check for the function

Comment (draage2572) [inline Command/GearmanJobExecuteCommand.php:82]: this only affects a subset of commands

File: Command/GearmanJobExecuteCommand.php [hunk 3]
```diff
@@ -79,6 +79,16 @@ protected function configure()
...
+            pcntl_async_signals(true);
+
+            pcntl_signal(SIGTERM, [$this->gearmanExecute, 'handleSystemSignal']);
```
Comment (draage2572) [inline Command/GearmanJobExecuteCommand.php:85]: should register signal handling on the executor itself

**Review by mihaileu — APPROVED** [review]

**Pull request merged**